src/status_im/transport/message/protocol.cljs

```diff
@@ -105,8 +105,15 @@
            this)
           (send-with-pubkey cofx params)))))
   (receive [this chat-id signature _ cofx]
-    (let [old-message (Message. (:text content) content-type message-type
-                                clock-value timestamp)]
+    (let [old-content      (if (or (= content-type constants/content-type-command)
+                                   (= content-type constants/content-type-command-request))
+                             (dissoc content :command-path)
+                             (:text content))
+          old-content-type (if (= ["request" #{:personal-chats}] (:command-path content))
```

Comment:
Could you add a comment about what this is about?

Reply:
It's about reconstructing the old content-type for message-id calculation, which is consistent with the old format.
It's necessary, because at the system boundary, we convert message with the old format to new format, therefore changing their content (and so id which is derived from sha of serialised message).
Unfortunately, calculation of that id happens **after** those coercions, that's why even the same messages can have different calculated id depending on version of app it's run on, leading to duplicated messages.
Best solution (besides changing the format of the message-id altogether) would be to calculate message id before any coercions take place, but that would require much more extensive code changes